tests/xgboost/test_xgboost_autolog.py

```diff
@@ -194,6 +194,57 @@ def test_xgb_autolog_sklearn():
     np.testing.assert_allclose(loaded_model.predict(X), model.predict(X))
 
 
+def test_xgb_autolog_with_sklearn_outputs_do_not_reflect_training_dataset_mutations():
+    original_xgb_regressor_fit = xgb.XGBRegressor.fit
+    original_xgb_regressor_predict = xgb.XGBRegressor.predict
+
+    try:
+
+        def patched_xgb_regressor_fit(self, *args, **kwargs):
+            X = args[0]
+            X["TESTCOL"] = 5
+            return original_xgb_regressor_fit(self, *args, **kwargs)
+
+        def patched_xgb_regressor_predict(self, *args, **kwargs):
+            X = args[0]
+            X["TESTCOL"] = 5
+            return original_xgb_regressor_predict(self, *args, **kwargs)
+
+        xgb.XGBRegressor.fit = patched_xgb_regressor_fit
+        xgb.XGBRegressor.predict = patched_xgb_regressor_predict
```

Comment:
```suggestion
    with mock.patch("xgboost.XGBRegressor.fit", patched_xgb_regressor_fit), mock.patch(
        "xgboost.XGBRegressor.predict", patched_xgb_regressor_predict
    ):
```

Can we use `mock.patch` and remove try-finally?

Reply:
Good catch! Done!